activesupport/lib/active_support/parameter_filter.rb

```diff
@@ -110,7 +110,14 @@ def value_for_key(key, value, parents = [], original_params = nil)
         elsif value.is_a?(Hash)
           value = call(value, parents, original_params)
         elsif value.is_a?(Array)
-          value = value.map { |v| v.is_a?(Hash) ? call(v, parents, original_params) : v }
```

Comment:
I _think_ we end up processing `Hash` values the same as before since recursively calling `value_for_key` already knows how to process `Hash` values (above on line 111).